ec/ecdatasource/deploymentdatasource/schema.go

```diff
@@ -18,103 +18,79 @@
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
 )
 
-func newSchema() map[string]*schema.Schema {
-	return map[string]*schema.Schema{
-		"alias": {
-			Type:     schema.TypeString,
-			Computed: true,
-		},
-		"healthy": {
-			Type:     schema.TypeBool,
-			Computed: true,
-		},
-		"id": {
-			Type:     schema.TypeString,
-			Required: true,
-		},
-		"name": {
-			Type:     schema.TypeString,
-			Computed: true,
-		},
-		"region": {
-			Type:     schema.TypeString,
-			Computed: true,
-		},
-		"deployment_template_id": {
-			Type:     schema.TypeString,
-			Computed: true,
-		},
-		"traffic_filter": {
-			Type:     schema.TypeList,
-			Computed: true,
-			Elem: &schema.Schema{
-				Type: schema.TypeString,
+func (d *DataSource) GetSchema(ctx context.Context) (tfsdk.Schema, diag.Diagnostics) {
+	return tfsdk.Schema{
+		Attributes: map[string]tfsdk.Attribute{
+			"alias": {
+				Type:        types.StringType,
+				Description: "Deployment alias.",
+				Computed:    true,
 			},
-		},
-		"observability": {
-			Type:     schema.TypeList,
-			Computed: true,
-			Elem:     newObservabilitySettings(),
-		},
-		"tags": {
-			Type:     schema.TypeMap,
-			Computed: true,
-			Elem: &schema.Schema{
-				Type: schema.TypeString,
+			"healthy": {
+				Type:        types.BoolType,
+				Description: "Overall health status of the deployment.",
+				Computed:    true,
 			},
-		},
-
-		// Deployment resources
-		"elasticsearch": {
-			Type:     schema.TypeList,
-			Computed: true,
-			Elem:     newElasticsearchResourceInfo(),
-		},
-		"kibana": {
-			Type:     schema.TypeList,
-			Computed: true,
-			Elem:     newKibanaResourceInfo(),
-		},
-		"apm": {
-			Type:     schema.TypeList,
-			Computed: true,
-			Elem:     newApmResourceInfo(),
-		},
-		"integrations_server": {
-			Type:     schema.TypeList,
-			Computed: true,
-			Elem:     newIntegrationsServerResourceInfo(),
-		},
-		"enterprise_search": {
-			Type:     schema.TypeList,
-			Computed: true,
-			Elem:     newEnterpriseSearchResourceInfo(),
-		},
-	}
-}
-
-func newObservabilitySettings() *schema.Resource {
-	return &schema.Resource{
-		Schema: map[string]*schema.Schema{
-			"deployment_id": {
-				Type:     schema.TypeString,
-				Computed: true,
+			"id": {
+				Type:        types.StringType,
+				Description: "The unique ID of the deployment.",
+				Required:    true,
+			},
+			"name": {
+				Type:        types.StringType,
+				Description: "The name of the deployment.",
+				Computed:    true,
+			},
+			"region": {
+				Type:        types.StringType,
+				Description: "Region where the deployment can be found.",
+				Computed:    true,
 			},
-			"ref_id": {
-				Type:     schema.TypeString,
-				Computed: true,
+			"deployment_template_id": {
+				Type:        types.StringType,
+				Description: "ID of the deployment template used to create the deployment.",
```

Comment:
```suggestion
				Description: "ID of the deployment template this deployment is based off.",
```

Deployment templates can be changed so we shouldn't talk about creation here.